src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs

```diff
@@ -7736,5 +7736,75 @@ class A {
                 format,
                 "delegate*<in Int32, ref readonly String>");
         }
+
+        [Theory, WorkItem(52490, "https://github.com/dotnet/roslyn/issues/52490")]
+        [InlineData("byte", "byte")]
+        [InlineData("byte", "System.Byte")]
+        [InlineData("sbyte", "sbyte")]
+        [InlineData("sbyte", "System.SByte")]
+        [InlineData("short", "short")]
+        [InlineData("short", "System.Int16")]
+        [InlineData("ushort", "ushort")]
+        [InlineData("ushort", "System.UInt16")]
+        // int is the default type and is not shown
+        [InlineData("uint", "uint")]
+        [InlineData("uint", "System.UInt32")]
+        [InlineData("long", "long")]
+        [InlineData("long", "System.Int64")]
+        [InlineData("ulong", "ulong")]
+        [InlineData("ulong", "System.UInt64")]
+        public void TestEnumWithUnderlyingType_ShowForNonDefaultTypes(string displayTypeName, string underlyingTypeName)
+        {
+            var text = @$"
+enum E : {underlyingTypeName}
+{{
+    A, B
+}}";
+
+            Func<NamespaceSymbol, Symbol> findSymbol = global =>
+                global.GetTypeMembers("E", 0).Single();
+
+            var format = new SymbolDisplayFormat(memberOptions: SymbolDisplayMemberOptions.IncludeType, kindOptions: SymbolDisplayKindOptions.IncludeTypeKeyword);
+
+            TestSymbolDescription(
+                text,
+                findSymbol,
+                format,
+                $"enum E : {displayTypeName}",
+                SymbolDisplayPartKind.Keyword,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.EnumName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.Punctuation,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.Keyword);
+        }
+
+        [Theory, WorkItem(52490, "https://github.com/dotnet/roslyn/issues/52490")]
+        [InlineData("")]
+        [InlineData(": int")]
+        [InlineData(": System.Int32")]
+        public void TestEnumWithUnderlyingType_DontShowForDefaultType(string defaultType)
+        {
+            var text = @$"
+enum E {defaultType}
+{{
+    A, B
+}}";
+
+            Func<NamespaceSymbol, Symbol> findSymbol = global =>
+                global.GetTypeMembers("E", 0).Single();
+
+            var format = new SymbolDisplayFormat(memberOptions: SymbolDisplayMemberOptions.IncludeType, kindOptions: SymbolDisplayKindOptions.IncludeTypeKeyword);
+
+            TestSymbolDescription(
+                text,
+                findSymbol,
+                format,
+                $"enum E",
+                SymbolDisplayPartKind.Keyword,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.EnumName);
+        }
```

Comment:
so this should only be shown in the case where we're checking on the top level type.  if the enum is referenced anywhere (like in a method signature) it woudl not be shown there.  Can you write tests showing that that's the case.

Also, we'd likely need this for VB.

Reply:
> so this should only be shown in the case where we're checking on the top level type

Yes. Adding the underlying type unconditionaly breaks all kinds of features (CI build failures confirm this). I haven't found an obvious way to prevent this. I think we need a new `formatOption`. But which one and where exactly should it go?

If such an option is present I could turn it of before any `VisitNamedType` calls e.g. if the SymbolDisplay is requested for an EnumMember.

Can you please give me some hints about what exactly you would expect here to solve this?

Reply:
> Yes. Adding the underlying type unconditionaly breaks all kinds of features

I think we should only doing this in the codepth that is also adding `enum`.  So we only would show: `enum E : byte`  not `E : byte` just because `E` was used in a reference location.  The SymbolDisplay code should already be partitioned to support this concept afaict.